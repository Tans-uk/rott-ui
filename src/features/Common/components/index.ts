export * from './CommonItem'
export * from './CommonItemContainer'
