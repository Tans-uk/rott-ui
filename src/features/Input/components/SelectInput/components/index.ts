export * from './SelectInput'
export * from './SelectInputModalComponent'
