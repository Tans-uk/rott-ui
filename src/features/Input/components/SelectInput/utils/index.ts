export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
