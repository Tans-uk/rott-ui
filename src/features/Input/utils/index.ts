export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
