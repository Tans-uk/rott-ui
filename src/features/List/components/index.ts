export * from './List'
export * from './FavoriteSwipeComponent'
export * from './ListSkeletonItem'
