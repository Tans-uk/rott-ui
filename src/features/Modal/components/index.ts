export * from './ModalComponent'
export * from './ModalContainer'
