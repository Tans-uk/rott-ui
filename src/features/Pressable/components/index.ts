export * from './Pressable'
export * from './PressableContainer'
