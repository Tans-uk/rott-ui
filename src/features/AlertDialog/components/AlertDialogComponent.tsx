import React, {useEffect, type FC} from 'react'

import {ActivityIndicator} from 'react-native'

import {AlertDialog} from '..'
import {formatMessage} from '../../../libs'
import {themeConfig} from '../../../providers'
import {display} from '../../../utils'
import {Button} from '../../Button'
import {EmptyState} from '../../EmptyState'
import {Icon} from '../../Icon'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {Separator} from '../../Separator'
import {type AlertDialogModel} from '../models'

export const AlertDialogComponent: FC<AlertDialogModel> = ({
  id,
  title,
  text,
  emptyState,
  icon,
  buttons = {
    cancelButton: {
      text: 'COMMON.CANCEL',
      variant: 'danger',
      onPress: undefined,
    },
    confirmButton: undefined,
  },
  showActivityIndicator,
  autoClose,
}) => {
  const hasMoreThanTwoButtons = Array.isArray(buttons)

  let autoCloseTimer: ReturnType<typeof setTimeout> | null = null
  useEffect(() => {
    if (autoClose) {
      autoCloseTimer = setTimeout(() => {
        AlertDialog.hide(id)
      }, autoClose)
    }

    return () => {
      if (autoCloseTimer) clearTimeout(autoCloseTimer)
    }
  }, [autoClose])

  return (
    <Item
      testID='alert-dialog-test-id'
      alignItemsCenter
      justifyContentCenter
      width={display.setWidth(80)}
      borderRadius={16}
      backgroundColor={themeConfig.colors['grey-100']}>
      {showActivityIndicator && (
        <Item marginTop={24}>
          <ActivityIndicator size='large' color={themeConfig.colors.primary} />
        </Item>
      )}

      {title && (
        <>
          <Item justifyContentCenter alignItemsCenter height={55}>
            <Label
              variant='grey-900'
              fontFamily='Markpro-Bold'
              textCenter
              fontSize='xl'
              fontWeight={700}>
              {title}
            </Label>
          </Item>

          <Separator orientation='horizontal' variant='neutral-alpha-200' size='full' />
        </>
      )}

      <Item
        size='full'
        paddingHorizontal={16}
        paddingTop={emptyState ? 0 : icon ? 24 : 16}
        paddingBottom={emptyState || icon ? 24 : 16}
        justifyContentCenter
        alignItemsCenter>
        {emptyState && (
          <EmptyState
            name={emptyState?.name ?? 'EMPTY_GENERAL_WARNING'}
            width={emptyState?.width ?? 152}
            height={emptyState?.width ?? 152}
          />
        )}

        {icon && (
          <Icon
            name={icon?.name ?? 'WARNING'}
            width={icon?.width ?? 64}
            height={icon?.height ?? 64}
            variant={icon.variant}
          />
        )}

        <Item
          size='full'
          justifyContentCenter
          alignItemsCenter
          marginTop={icon ? 8 : 0}
          marginHorizontal={16}>
          <Label variant='grey-900' fontFamily='Markpro' textCenter fontSize='lg' fontWeight={500}>
            {text}
          </Label>
        </Item>
      </Item>

      <Separator orientation='horizontal' variant='neutral-alpha-200' size='full' />

      {buttons && (
        <Item row={!hasMoreThanTwoButtons} size='full' justifyContentCenter alignItemsCenter>
          {!hasMoreThanTwoButtons && buttons?.cancelButton && (
            <Button
              testID={buttons?.cancelButton?.testID || 'alert-dialog-cancel-button-test-id'}
              variant='transparent'
              color={buttons?.cancelButton?.variant || 'danger'}
              fontSize='lg'
              fontWeight={500}
              onPress={() => {
                !!buttons?.cancelButton?.onPress && buttons.cancelButton.onPress()

                AlertDialog.hide(id)
              }}>
              {formatMessage(buttons.cancelButton.text)}
            </Button>
          )}

          {!hasMoreThanTwoButtons && buttons?.confirmButton && (
            <Separator width={1} orientation='vertical' height={55} variant='neutral-alpha-200' />
          )}

          {!hasMoreThanTwoButtons && buttons?.confirmButton && (
            <Button
              testID={buttons?.confirmButton?.testID || 'alert-dialog-confirm-button-test-id'}
              variant='transparent'
              color={buttons?.confirmButton?.variant || 'danger'}
              fontSize='lg'
              fontWeight={500}
              onPress={() => {
                !!buttons?.confirmButton?.onPress && buttons.confirmButton.onPress()
              }}>
              {formatMessage(buttons.confirmButton.text)}
            </Button>
          )}

          {hasMoreThanTwoButtons &&
            buttons &&
            buttons
              .filter((button) => !!button)
              .map((button, index) => {
                const {text: buttonText, variant: buttonVariant, onPress: buttonOnPress} = button

                return (
                  <Item key={`button-${text?.toSeoFriendly()}-${index}`}>
                    <Item row height={56}>
                      <Button
                        variant='transparent'
                        color={buttonVariant}
                        fontSize='lg'
                        fontWeight={500}
                        onPress={() => {
                          !!buttonOnPress && buttonOnPress()

                          AlertDialog.hide(id)
                        }}>
                        {formatMessage(buttonText)}
                      </Button>
                    </Item>

                    {buttons.length - 1 !== index && (
                      <Separator orientation='horizontal' size='full' variant='neutral-alpha-200' />
                    )}
                  </Item>
                )
              })}
        </Item>
      )}
    </Item>
  )
}
