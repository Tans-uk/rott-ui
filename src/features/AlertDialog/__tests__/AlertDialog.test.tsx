import {render} from '../../../__tests__/utils/testUtils'
import {formatMessage} from '../../../libs'
import {AlertDialogComponent} from '../components/AlertDialogComponent'
import type {AlertDialogProps} from '../models'

const mockLessThanOrEqualTwoButtons: AlertDialogProps = {
  title: formatMessage('TEST.WITH.PARAM', {testText: 'Test Title'}),
  text: formatMessage('TEST.WITH.PARAM', {testText: 'Test Text'}),
  icon: {
    name: 'QR_TRANSFER',
    width: 64,
    height: 64,
    variant: 'success',
  },
  buttons: {
    cancelButton: {
      text: 'COMMON.CANCEL',
      onPress: jest.fn(),
      variant: 'danger',
    },
    confirmButton: {
      text: 'COMMON.OK',
      onPress: jest.fn(),
      variant: 'primary',
    },
  },
  showActivityIndicator: true,
  visible: true,
  testID: 'alert-dialog-test-id',
}

const mockMoreThanTwoButtons: AlertDialogProps = {
  title: formatMessage('TEST.WITH.PARAM', {testText: 'Test Title'}),
  text: formatMessage('TEST.WITH.PARAM', {testText: 'Test Text'}),
  icon: {
    name: 'QR_TRANSFER',
    width: 64,
    height: 64,
    variant: 'success',
  },
  buttons: [
    {
      text: 'COMMON.CANCEL',
      onPress: jest.fn(),
      variant: 'danger',
    },
    {
      text: 'COMMON.OK',
      onPress: jest.fn(),
      variant: 'primary',
    },
    {
      text: 'COMMON.BACK',
      onPress: jest.fn(),
      variant: 'primary',
    },
  ],
  showActivityIndicator: true,
  visible: true,
  testID: 'alert-dialog-test-id',
}

describe('AlertDialogComponent', () => {
  const testId = {
    alertDialogTestId: 'alert-dialog-test-id',
  }

  it('should match snapshot on first render', () => {
    const {alertDialogTestId} = testId

    const renderedAlertDialog = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockLessThanOrEqualTwoButtons} />
    )

    expect(renderedAlertDialog.toJSON()).toMatchSnapshot()
  })

  it('title ve text ekranda düzgün şekilde renderlanmalı', () => {
    const {alertDialogTestId} = testId
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockLessThanOrEqualTwoButtons} />
    )

    const title = getByText(formatMessage('TEST.WITH.PARAM', {testText: 'Test Title'}), {
      exact: true,
    })
    const text = getByText(formatMessage('TEST.WITH.PARAM', {testText: 'Test Text'}), {
      exact: true,
    })

    expect(title).toBeOnTheScreen()
    expect(text).toBeOnTheScreen()
  })

  it('iki button tanımlı ise alert dialog doğru renderlanmalı', () => {
    const {alertDialogTestId} = testId
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockLessThanOrEqualTwoButtons} />
    )

    const confirmButton = getByText(formatMessage('COMMON.OK'))
    const cancelButton = getByText(formatMessage('COMMON.CANCEL'))

    expect(confirmButton).toBeOnTheScreen()
    expect(cancelButton).toBeOnTheScreen()
  })

  it('iki veya daha fazla button tanımlı ise alert dialog doğru renderlanmalı', () => {
    const {alertDialogTestId} = testId
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockMoreThanTwoButtons} />
    )

    const confirmButton = getByText(formatMessage('COMMON.OK'))
    const cancelButton = getByText(formatMessage('COMMON.CANCEL'))
    const backButton = getByText(formatMessage('COMMON.BACK'))

    expect(confirmButton).toBeOnTheScreen()
    expect(cancelButton).toBeOnTheScreen()
    expect(backButton).toBeOnTheScreen()
  })
})
