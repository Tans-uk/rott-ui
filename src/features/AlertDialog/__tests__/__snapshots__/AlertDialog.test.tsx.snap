// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialogComponent should match snapshot on first render 1`] = `
<KeyboardProvider
  enabled={true}
>
  <View
    flex={1}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      alignItemsCenter={true}
      backgroundColor="rgba(234, 239, 240, 1)"
      borderRadius={16}
      justifyContentCenter={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(234, 239, 240, 1)",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "center",
          "width": 600,
        }
      }
      testID="alert-dialog-test-id"
      width={312}
    >
      <View
        marginTop={24}
        style={
          {
            "flex": undefined,
            "flexDirection": "column",
            "marginTop": 46,
          }
        }
      >
        <ActivityIndicator
          color="rgba(0, 169, 206, 1)"
          size="large"
        />
      </View>
      <View
        alignItemsCenter={true}
        height={55}
        justifyContentCenter={true}
        style={
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "column",
            "height": 106,
            "justifyContent": "center",
            "maxHeight": 106,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(34, 63, 70, 1)",
              "fontFamily": "Markpro-Bold",
              "fontSize": 18,
              "fontWeight": undefined,
              "textAlign": "center",
            }
          }
        >
          TEST WITH Test Title
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "rgba(34, 63, 70, 0.15)",
            "height": 2,
            "maxHeight": 2,
            "opacity": 1,
            "width": "100%",
          }
        }
        variant="neutral-alpha-200"
      />
      <View
        alignItemsCenter={true}
        justifyContentCenter={true}
        paddingBottom={24}
        paddingHorizontal={16}
        paddingTop={24}
        style={
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingBottom": 46,
            "paddingHorizontal": 31,
            "paddingTop": 46,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "flex": undefined,
              "flexDirection": "column",
            }
          }
        >
          <Text
            fill="rgba(63, 182, 24, 1)"
            height={123}
            strokeWidth={0}
            testID="mock-icon"
            width={123}
          >
            MockIcon
          </Text>
        </View>
        <View
          alignItemsCenter={true}
          justifyContentCenter={true}
          marginHorizontal={16}
          marginTop={8}
          style={
            {
              "alignItems": "center",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginHorizontal": 31,
              "marginTop": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgba(34, 63, 70, 1)",
                "fontFamily": "Markpro",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "center",
              }
            }
          >
            TEST WITH Test Text
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "rgba(34, 63, 70, 0.15)",
            "height": 2,
            "maxHeight": 2,
            "opacity": 1,
            "width": "100%",
          }
        }
        variant="neutral-alpha-200"
      />
      <View
        alignItemsCenter={true}
        justifyContentCenter={true}
        style={
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "white",
              "borderRadius": 16,
              "borderWidth": undefined,
              "color": "danger",
              "flex": 1,
              "flexDirection": "row",
              "height": 108,
              "justifyContent": "center",
              "opacity": 1,
              "position": "relative",
              "width": 658,
            }
          }
          testID="alert-dialog-cancel-button-test-id"
        >
          <Text
            numberOfLines={1}
            style={
              {
                "color": "rgba(246, 83, 83, 1)",
                "flexShrink": 1,
                "fontFamily": "Markpro-Book",
                "fontSize": 26,
                "fontWeight": 500,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "rgba(34, 63, 70, 0.15)",
              "height": 106,
              "maxHeight": 106,
              "opacity": 1,
              "width": 2,
            }
          }
          variant="neutral-alpha-200"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "white",
              "borderRadius": 16,
              "borderWidth": undefined,
              "color": "primary",
              "flex": 1,
              "flexDirection": "row",
              "height": 108,
              "justifyContent": "center",
              "opacity": 1,
              "position": "relative",
              "width": 658,
            }
          }
          testID="alert-dialog-confirm-button-test-id"
        >
          <Text
            numberOfLines={1}
            style={
              {
                "color": "rgba(0, 169, 206, 1)",
                "flexShrink": 1,
                "fontFamily": "Markpro-Book",
                "fontSize": 26,
                "fontWeight": 500,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            Ok
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "flexDirection": "column-reverse",
            "justifyContent": "flex-start",
            "top": 30,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "bottom": 40,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
  </View>
</KeyboardProvider>
`;
